lib/WWW/Mechanize/Cached.pm

```diff
@@ -101,11 +103,25 @@ around _make_request => sub {
     $response->decode();
     delete $response->{handlers};    ## no critic
 
-    $self->cache->set( $req, nfreeze($response) ) if $should_cache;
+    if ($should_cache) {
+        $self->_last_request($req);
+        $self->cache->set( $req, nfreeze($response) );
+    }
 
     return $response;
 };
 
+sub clear_cache {
```

Comment:
Naming could be much improved in the whole module, but backwards compatibility... :-( e.g. `is_cached` is false when the current page *is* cached, but wasn't loaded from a cache. I'll think of  a better name of the new method, though.

Reply:
What about `uncache_current`? Or `forget_current`? Or `_latest` instead of `current`?

`remove_current_request_from_cache` seems to long and Javasque to me.

Reply:
Maybe `invalidate_last_request` or `invalidate_latest_request`?  Current makes me think the request is still in progress.

Alternatively, if this were more generic and allowed the invalidation of any request it could just be `invalidate_request( $request)`.

Reply:
> Naming could be much improved in the whole module, but backwards compatibility

Agreed.  This module goes back a long time.  The original author passed away and it has changed hands a couple of times since then, but it's still quite useful.

Reply:
Using $request as a parameter would be nice, but if I understand `_make_request` correctly, it sometimes modifies the request before caching it.